src/Orleans/Runtime/Constants.cs

```diff
@@ -59,7 +59,8 @@ internal class Constants
         public static readonly GrainId TypeManagerId = GrainId.GetSystemTargetGrainId(17);
         public static readonly GrainId ProviderManagerSystemTargetId = GrainId.GetSystemTargetGrainId(19);
         public static readonly GrainId DeploymentLoadPublisherSystemTargetId = GrainId.GetSystemTargetGrainId(22);
-       
+        public static readonly GrainId ClusterDirectoryServiceId = GrainId.GetSystemTargetGrainId(24);
```

Comment:
you need to add this new ClusterDirectoryServiceId  in the singletonSystemTargetNames  below.